x/market/client/cli/flags.go

```diff
@@ -0,0 +1,45 @@
+package cli
+
+import (
+	"bytes"
+	"encoding/json"
+	flag "github.com/spf13/pflag"
+)
+
+const (
+	FlagDealFile = "deal-json-file"
+)
+
+type createDealInputs struct {
+	DataSchema            []string `json:"data_schema"`
+	Budget                string   `json:"budget"`
+	WantDataCount         uint64   `json:"want_data_count"`
+	TrustedDataValidators []string `json:"trusted_data_validators"`
+}
```

Comment:
How about using `MsgCreateDeal` in `proto/panacea/market/v2/tx.proto`.

If Budget is a problem, it seems good to change a spec.

And you can refer to the `json_name` option of  `did/did.proto`

Reply:
IMHO, it's ok to have this separated `createDealInputs` struct, because this struct is slightly different from `MsgCreateDeal`. I don't think it's good to modify the `tx.proto` for adding some json_name tags.

Also, `MsgCreateDeal` can be more complex in the future, as [MsgCreatePool](https://github.com/osmosis-labs/osmosis/blob/ba4dc51cc8e5c45e96fa14fd4b5ca9321ca92a78/x/gamm/types/tx.pb.go#L34) in the Osmosis.

Reply:
Aha, It's ok.
Let's leave this struct as it is.